tests/mdns_test.rs

```diff
@@ -680,6 +680,81 @@ fn service_with_ipv4_only() {
     d.shutdown().unwrap();
 }
 
+#[test]
+fn test_disable_interface_cache() {
+    // Create a server
+    let server = ServiceDaemon::new().expect("Failed to create the server");
+
+    // Register a service with one IPv4.
+    let ty_domain = "_disable-intf._tcp.local.";
+    let now = SystemTime::now()
+        .duration_since(SystemTime::UNIX_EPOCH)
+        .unwrap();
+    let instance_name = now.as_micros().to_string();
+    let service_ip_addr = my_ip_interfaces()
+        .iter()
+        .find(|iface| iface.ip().is_ipv4())
+        .map(|iface| iface.ip())
+        .unwrap();
+
+    let host_name = "disabled_intf_host.local.";
+    let port = 5201;
+    let my_service = ServiceInfo::new(
+        &ty_domain,
+        &instance_name,
+        host_name,
+        &service_ip_addr,
+        port,
+        None,
+    )
+    .expect("Invalid service info");
+    server
+        .register(my_service)
+        .expect("Failed to register our service");
+
+    // Create a client
+    let client = ServiceDaemon::new().expect("Failed to create the client");
+
+    // Give it some time to cache mDNS records.
+    sleep(Duration::from_secs(1));
+
+    // Disable the interface for the client.
+    client.disable_interface(service_ip_addr).unwrap();
+
+    // Browse for the service.
+    let handle = client.browse(&ty_domain).unwrap();
+    let timeout = Duration::from_secs(1);
+    let mut resolved = false;
+
+    // run till timeout and it should not resolve.
+    loop {
```

Comment:
Out of curiosity, what's the main purpose of this loop? I cannot figure it out. Is there the possibility that `recv_timeout` does not close after the `timeout` is expired?

Reply:
`loop` is needed because `recv_timeout` could first return events other than `ServiceResolved` ( for example `ServiceFound`) , but we have to wait for `ServiceResolved` or a timeout error.

Reply:
Ah, I see now, thanks for your explanation!